services/prometheus/prometheus.go

```diff
@@ -376,8 +380,12 @@ func (svc *Service) saveConfigAndReload(cfg []byte) error {
 		return errors.WithStack(err)
 	}
 
+	alertManagerRules, err := svc.alertManager.ReadRules()
+	if err != nil {
+		svc.l.Warnf("Cannot load Alert Manager rules: %s", err)
+	}
 	// compare with new config
-	if reflect.DeepEqual(cfg, oldCfg) {
+	if reflect.DeepEqual(cfg, oldCfg) && alertManagerRules == svc.cachedAlertManagerRules {
```

Comment:
If the config hasn't change but it is not cached:
1. Is that possible?
2. Shouldn't we only cache the config **without** restarting Prometheus?